sdk/keyvault/azure-security-keyvault/tests/test_secrets_client.py

```diff
@@ -227,52 +210,74 @@ def test_backup_restore(self, vault_client, **kwargs):
 
     @ResourceGroupPreparer()
     @VaultClientPreparer(enable_soft_delete=True)
-    def test_recover_purge(self, vault_client, **kwargs):
+    def test_recover(self, vault_client, **kwargs):
         self.assertIsNotNone(vault_client)
         client = vault_client.secrets
 
         secrets = {}
 
         # create secrets to recover
-        for i in range(0, self.list_test_size):
-            secret_name = self.get_resource_name("secrec{}".format(str(i)))
-            secret_value = self.get_resource_name("secval{}".format(str(i)))
-            secrets[secret_name] = client.set_secret(secret_name, secret_value)
-
-        # create secrets to purge
-        for i in range(0, self.list_test_size):
-            secret_name = self.get_resource_name("secprg{}".format(str(i)))
-            secret_value = self.get_resource_name("secval{}".format(str(i)))
+        for i in range(self.list_test_size):
+            secret_name = "secret{}".format(i)
+            secret_value = "value{}".format(i)
             secrets[secret_name] = client.set_secret(secret_name, secret_value)
 
         # delete all secrets
         for secret_name in secrets.keys():
             client.delete_secret(secret_name)
-
-        if not self.is_playback():
-            time.sleep(20)
+        for secret_name in secrets.keys():
+            self._poll_until_no_exception(
+                functools.partial(client.get_deleted_secret, secret_name), ResourceNotFoundError
+            )
 
         # validate all our deleted secrets are returned by list_deleted_secrets
         deleted = [s.name for s in client.list_deleted_secrets()]
-
         self.assertTrue(all(s in deleted for s in secrets.keys()))
 
         # recover select secrets
-        for secret_name in [s for s in secrets.keys() if s.startswith("secrec")]:
+        for secret_name in secrets.keys():
             client.recover_deleted_secret(secret_name)
 
-        # purge select secrets
-        for secret_name in [s for s in secrets.keys() if s.startswith("secprg")]:
-            client.purge_deleted_secret(secret_name)
+        # validate the recovered secrets exist
+        for secret_name in secrets.keys():
+            secret = self._poll_until_no_exception(
+                functools.partial(client.get_secret, secret_name), ResourceNotFoundError
+            )
+            self._assert_secret_attributes_equal(secret, secrets[secret.name])
 
-        if not self.is_playback():
-            time.sleep(20)
+    @ResourceGroupPreparer()
+    @VaultClientPreparer(enable_soft_delete=True)
+    def test_purge(self, vault_client, **kwargs):
```

Comment:
do we want to still keep `**kwargs` ?

Reply:
Maybe not but it doesn't affect test reliability so I'll leave it for future consideration.